BUGFIX: Fixed duplicate Home links appearing

UndefinedOffset opened the pull request:
Due to the location of the "Home" link for every module that the documentation covers it creates a duplicate link labeled "Home" this pull request moves the "Home" link outside the Menu loop which creates it only once with a link to the docs home.

dhensby commented on templates/Includes/DocumentationSidebar.ss:
```diff
@@ -3,9 +3,9 @@
+			<li><a href="$Link" class="top">Home</a></li>
 
 			<% loop Menu %>
-				<li><a href="$Link" class="top">Home</a></li>
```
Won't this mean there's no link to the root of each of the menus?

UndefinedOffset replied:
I'll double check again in about an hour but in my testing it would put say "Home" followed right below by say "doc viewer" then home again and the next module.

UndefinedOffset replied:
With both home and the module name linking to that modules docs

UndefinedOffset commented:
@dhensby Based on your line comments here's what I'm seeing using a clean install of 3.2 and the latest master of docviewer plus a composer.lock for a project I'm working on. Looking at this template closer now I think you are correct to point out that it should cause what you are saying, it shouldn't have a top level link. I'm guessing the reason it does is quite possibly coming from DocumentationViewer::getMenu() more than anything, it's almost like the first Children loop is redundant. So it's iterating through Menu only to iterate through the first level of Children as the actual menu list. I think I'll spend a bit more time with this and see if I can figure out why that is the case and make some adjustments to DocumentationViewer::getMenu() and probably a few other areas to address this more completely. I've still attached two images of what I'm seeing and what my original commit addresses. As well I may have one more issue to fix up unrelated to this, want me to just include that in this pull (if it turns out to actually be a problem) or open a new for that one?

__Before Change:__
![before-pull-99](https://cloud.githubusercontent.com/assets/1391558/13253409/4af9515a-da12-11e5-9259-4566d69024e5.jpg)

__After Change__
![after-pull-99](https://cloud.githubusercontent.com/assets/1391558/13253487/a9a9c518-da12-11e5-89b9-f5cc226615e3.jpg)

UndefinedOffset commented:
@dhensby I think I see what the original intention of this was, when there is a default entity "Home" actually links to the root of that entity/module. So the home link should have really been under the if DefaultEntity check. But that does make me wonder if there is no default entity should we still have a home link before the loop Menu?

dhensby commented:
Having the link "home" repeatedly is confusing.

I think you're right with the intended purpose, we need to make sure we're not breaking anything with this going forward

UndefinedOffset commented:
@dhensby I've adjusted my commit to have a default home link if there is no default entity and moved the other home link for if there is a default entity under the check for DefaultEntity. This still produces the result in my second screenshot if there is no default entity, but produces a home link that points to the root of the default entity if there is one. So you end up with the below if there is a default entity.

![after-pull-99-b](https://cloud.githubusercontent.com/assets/1391558/13256871/c962fa72-da22-11e5-9b64-d093138d2a3a.jpg)

dhensby commented:
OK - I'd like a second opinion on this as I'm not that familiar with the modules. thoughts @camfindlay?

camfindlay commented:
@wilr ? Actually, what I might look to do is test this against our community documentation websites

camfindlay commented:
Also @UndefinedOffset might be good if you can check out what we're doing with docsviewer over at https://github.com/silverstripe/doc.silverstripe.org/blob/master/app/_config/docsviewer.yml

Maybe we're doing it wrong, note we're using DefaultEntity on all our versions I never did find out why.

UndefinedOffset commented:
@camfindlay I believe the DefaultEntity thing focuses the navigation, so in the case of doc.silverstripe.org it keeps the tree focused on just for example the 3.3 documentation when you are in that url. If you look at the screenshots (the before) when there is no DefaultEntity and you are on the root of the documentation it shows all modules as the top level navigation. Where as when you are in the DefaultEntity it only shows that tree I believe.

camfindlay commented:
Testing this today with a docs site so far so good. If site is all still ok will merge this.

camfindlay commented:
Seems to be all working on a test docs site. Happy to merge.

camfindlay commented:
@UndefinedOffset would this be considers a bug fix or enhanced feature? Just working out whether I should release a new minor or patch version of the module today. Let me know.

camfindlay commented:
I'm going to do a new minor release, looking at diff there has been a few new public APIs introduced alongside the bug fixes.
